compiler/sigmatch.nim

```diff
@@ -2233,8 +2233,13 @@ proc paramTypesMatch*(m: var TCandidate, f, a: PType,
         # XXX this is still all wrong: (T, T) should be 2 generic matches
         # and  (int, int) 2 exact matches, etc. Essentially you cannot call
         # typeRel here and expect things to work!
-        let r = typeRel(z, f, arg[i].typ)
+        var r = typeRel(z, f, arg[i].typ)
         incMatches(z, r, 2)
+        if r == isNone:
+          if userConvMatch(c, m, f, a, arg) == nil and f.kind == tyVarargs:
+            if f.n == nil:
+              if typeRel(m, base(f), a) == isGeneric:
+                r = typeRel(m, base(f), a)
```

Comment:
Taken from https://github.com/nim-lang/Nim/blob/33cd883e27f7ba7564cbeb59d2d38fe02bb507bf/compiler/sigmatch.nim#L2174-L2182

Could be more cases should be handled